### ISSUE TITLE
uMatrix prevents opening blocked image

### ISSUE THREAD

**Issue by rekcuFniarB** [uBlockOrigin/uMatrix-issues#96]
### Prerequisites

- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
    - [x] I also searched the existing issues at <https://github.com/gorhill/uMatrix/issues>
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uMatrix](https://old.reddit.com/r/uMatrix/)
- I tried to reproduce the issue when...
    - [x] uMatrix is the only extension
    - [x] uMatrix with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uMatrix
- [x] I checked the [documentation](https://github.com/gorhill/uMatrix/wiki) to understand that the issue I report is not a normal behavior
- [x] I used the logger to rule out that the issue is caused by my ruleset

### Description

For some sites I have a rule blocking images from the same domain. 

![example screenshot](https://i.imgur.com/lOvU4Uj.png)

But sometimes I need to view one of blocked image. In previous version I could just open an image from context menu (i.e. right click on blocked image placeholder → view image):

![screenshot](https://i.imgur.com/2yf6ioF.png)

 But now uMatrix prevents opening it, showing internal error page:
![screenshot](https://i.imgur.com/LTKx5Lb.png)

There should be an option to bypass static files when they are opened directly.

### A specific URL where the issue occurs

Any URL

### Steps to Reproduce

1. Create a rule blocking images from the same domain for specific page.
2. Try to open any blocked image from context menu.

### Your environment

* uMatrix version: 1.3.16
* Browser Name and version: Firefox 63.0
* Operating System and version: Ubuntu 18.04

**Comment by uBlock-user** [comment]
You didn't post your ruleset as asked. Post it here.

**Comment by rekcuFniarB** [comment]
### Permanent rules:
```
https-strict: behind-the-scene false
matrix-off: about-scheme true
matrix-off: behind-the-scene true
matrix-off: chrome-extension-scheme true
matrix-off: chrome-scheme true
matrix-off: moz-extension-scheme true
matrix-off: opera-scheme true
matrix-off: vivaldi-scheme true
matrix-off: wyciwyg-scheme true
noscript-spoof: * true
referrer-spoof: * true
referrer-spoof: behind-the-scene false
* * * block
* * frame block
* 1st-party * allow
* 1st-party frame allow
linux.org.ru 1st-party image block
```

### Temporary rules
```
https-strict: behind-the-scene false
matrix-off: about-scheme true
matrix-off: behind-the-scene true
matrix-off: chrome-extension-scheme true
matrix-off: chrome-scheme true
matrix-off: moz-extension-scheme true
matrix-off: opera-scheme true
matrix-off: vivaldi-scheme true
matrix-off: wyciwyg-scheme true
noscript-spoof: * true
referrer-spoof: * true
referrer-spoof: behind-the-scene false
* * * block
* * frame block
* 1st-party * allow
* 1st-party frame allow
github.com githubassets.com * allow
linux.org.ru 1st-party image block
```

**Comment by uBlock-user** [comment]
Not sure if this is a bug or by-design.

**Comment by uBlock-user** [comment]
Not a bug, see - https://github.com/uBlockOrigin/uMatrix-issues/issues/59

**Comment by rekcuFniarB** [comment]
> Not a bug, see - #59, the new behaviour is a result of a fix applied to the existing behaviour which was considered as a bug.

OK, thanks. But it will be nice to have an option in settings to disable such behaviour.

**Comment by gorhill** [comment]
> But it will be nice to have an option in settings to disable such behaviour.

I will consider it.

**Comment by klopsi** [comment]
This is highly problematic.

I get many, many direct links to images to different sites.   I'm not going to unblock every direct image link.

**Comment by gorhill** [comment]
It's why there need to be more people using the dev build, to provide such feedback before the change makes it to stable release. I was skeptical about changing the behavior, and now I see the skepticism confirmed. With feedback, I would have make the change opt-in.

**Comment by gorhill** [comment]
@klopsi So just that I understand... You are blocking all images everywhere by default?

**Comment by klopsi** [comment]
Yes everything's blocked by default. Can I unblock all images from same domain?  Will donate.

**Comment by gorhill** [comment]
Right, my mistake, `* 1st-party image allow` is not a default rule. Does it solve the issue here if you add it?

**Comment by klopsi** [comment]
I don't see how to "Add a rule" in the Rules tab of uMatrix config.

**Comment by gorhill** [comment]
Just start to type in the right-hand pane, it's fully editable. Insert a blank line than paste the rule. You can also do it from the popup panel by selecting the global scope and allowing `image` (persist if it works).

**Comment by klopsi** [comment]
Ok added new rule.  Clicked Save and Commit.  Still have the problem.   Need to reload firefox?

**Comment by gorhill** [comment]
Alright I was in the middle of something else hence why I didn't try it myself.

So I just try `* 1st-party image allow` and it works fine on my side. When this does not work, try bypassing the cache -- hold shift or ctrl while clicking reload button.

**Comment by gorhill** [comment]
Dev builds are released in [Releases](https://github.com/gorhill/uMatrix/releases). They are labelled "Pre-release" (red).

**Comment by uBlock-user** [comment]
Latest one - https://github.com/gorhill/uMatrix/releases/tag/1.3.15rc2

**Comment by rekcuFniarB** [comment]
> Right, my mistake, * 1st-party image allow is not a default rule. Does it solve the issue here if you add it?

In my case no.

**Comment by RadoQS** [comment]
Back in the "good old days" firefox (or was it netscape) had content-control for images like for cookies: none, local-only, 3rd party. When images were disabled, placeholders were put, and via context-menu you could "load image in place" as if hadn't been blocked. If it was used elsewhere (on same page/tab or any other), it was filled-in everywhere.
I assume the "replace everywhere" is hard to implement, but the "in-place" would be awesome.
Is this the right place to voice the request, or should I create a new item?
Because the OP was about the basic functionality to allow exception from the block, while mine is just on-top of this a usability enhancement.

**Comment by sebast889** [comment]
Agree with @rekcuFniarB. Advanced setting or option to change this behavior would be welcome.
